tests/e2e/modelregistry_test.go

```diff
@@ -134,3 +136,54 @@ func (tc *ModelRegistryTestCtx) retrieveModelRegistry() *componentApi.ModelRegis
 
 	return mri
 }
+
+// ValidateNamespaceErrorReporting tests how operator reports validation errors in status conditions.
+func (tc *ModelRegistryTestCtx) ValidateNamespaceErrorReporting(t *testing.T) {
+	t.Helper()
+
+	t.Log("Testing operator validation error reporting")
+
+	originalDSC := tc.FetchDataScienceCluster()
+
+	// Restore original DSC state after the test
+	defer func() {
+		t.Log("Restoring original DSC state")
+		tc.EnsureResourceCreatedOrUpdated(
+			WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+			WithMutateFunc(testf.TransformPipeline(
+				testf.Transform(`.spec.components.modelregistry.managementState = "%s"`, originalDSC.Spec.Components.ModelRegistry.ManagementState),
+				testf.Transform(`.spec.components.modelregistry.registriesNamespace = "%s"`, originalDSC.Spec.Components.ModelRegistry.RegistriesNamespace),
+			)),
+		)
+	}()
+
+	t.Log("Creating DSC with ModelRegistry using an invalid namespace that operator will reject during reconciliation")
+	tc.EnsureResourceCreatedOrUpdated(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithMutateFunc(testf.TransformPipeline(
+			testf.Transform(`.spec.components.modelregistry.managementState = "Managed"`),
+			testf.Transform(`.spec.components.modelregistry.registriesNamespace = "invalid--namespace"`),
+		)),
+	)
+
+	t.Log("Checking that DSC reports ModelRegistryReady as False when ModelRegistry has validation errors")
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithCondition(And(
+			// Verify ModelRegistry is reported as not ready
+			jq.Match(`.status.conditions[] | select(.type == "ModelRegistryReady") | .status == "False"`),
+			// Verify error message is descriptive
+			jq.Match(`.status.conditions[] | select(.type == "ModelRegistryReady") | .message != null`),
```

Comment:
I think we could assert on the content of the message as well to verify it is really descriptive.